### PR TITLE
Fixed solar assemblies not dropping into the solar panel on construct…

### PR DESCRIPTION
**Pull request by xxalpha**
…ion.

Fixes https://github.com/tgstation/-tg-station/issues/10555

E: Fixes https://github.com/tgstation/-tg-station/issues/6839

### CODE REVIEW

File: code/modules/mob/inventory.dm [hunk 1]
```diff
@@ -114,6 +114,8 @@
 
 //Here lie drop_from_inventory and before_item_take, already forgotten and not missed.
 
+/mob/proc/can_drop_items()
+	return loc.allow_drop()
```
Comment (phil235) [inline code/modules/mob/inventory.dm]: a whole new proc just for a single fix?

File: code/modules/power/solar.dm [hunk 2]
```diff
@@ -238,17 +238,21 @@
...
 		if(istype(W, /obj/item/stack/sheet/glass) || istype(W, /obj/item/stack/sheet/rglass))
 			var/obj/item/stack/sheet/S = W
-			if(S.use(2))
-				glass_type = W.type
-				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
-				user.visible_message("[user] places the glass on the solar assembly.", "<span class='notice'>You place the glass on the solar assembly.</span>")
-				if(tracker)
-					new /obj/machinery/power/tracker(get_turf(src), src)
+			if(user.can_drop_items())
+				if(user.unEquip(src) && S.use(2))
```
Comment (phil235) [inline code/modules/power/solar.dm]: if unEquip() fails, then the user gets the message "You need two sheets of glass to put them ..."

**Comment by phil235** [comment]
There's a simpler way to fix it, just check that the solar assembly is anchored. It makes sense and fixes the issue. Also the whole wrench section of the attackby could be reworked.

Just have the isturf(loc) check before wrench/glass sheet action shit, bam, problem solved.

File: code/modules/power/solar.dm (hunk 2 above)
Comment (xxalpha) [inline code/modules/power/solar.dm]: It's not going to fail if it passes the first check, the unEquip is just to drop the assembly, as using drop_item would drop the glass.

Comment (phil235) [inline code/modules/power/solar.dm]: The UnEquip can fail. and then you'd get the message about not having enough glass despite having enough.

**Comment by phil235** [comment]
why close it?

**Pull request merged**